packages/backend/src/services/PromoteService/PromoteService.ts

```diff
@@ -798,95 +1176,30 @@ export class PromoteService extends BaseService {
             );
 
             // at this point, all permisions checks are done, so we can safely promote the dashboard and charts.
-            const newSpaces = await this.createNewSpaces(
+            // And return the list of dashboards and charts with the new space
+            promotionChanges = await this.createNewSpaces(
```

Comment:
SImplify code by performing actions based on a  mutable `promotionChanges` which is updated with the right uuids when stuff is created.